packages/graph-engine/src/programmatic/input.ts

```diff
@@ -59,7 +59,27 @@ export class Input<T = any> extends Port<T> {
       }
       //Otherwise do a structural comparison
       else if (JSON.stringify(this._type) !== JSON.stringify(opts.type)) {
-        this._dynamicType = opts.type;
+        const {variadic, _dynamicType } = this;
+
+        // for variadic ports, we need to set the dynamic type depending on all the other connected types
+        if (variadic && _dynamicType && JSON.stringify(_dynamicType) !== JSON.stringify(opts.type)) {          
+          const graph = this.node.getGraph();
+          const inEdges = graph.inEdges(this.node.id);
```

Comment:
This looks wrong, we're mapping over all edges, likely we want to filter on the port that is being connected

Reply:
fixed